app/controllers/bookmarks_controller.rb

```diff
@@ -4,15 +4,23 @@ class BookmarksController < ApplicationController
   requires_login
 
   def create
-    params.require(:post_id)
+    if params[:post_id].blank?
+      if params[:topic_id].blank?
+        raise Discourse::InvalidParameters.new("Either a post_id or a topic_id is required.")
+      end
+      topic_id = params[:topic_id].to_i
+    else
+      post_id = params[:post_id]&.to_i
```

Comment:
```suggestion
      post_id = params[:post_id].to_i
```

I don't think we need the lonely operator here since we've already checked that the param is not blank.

Reply:
:+1: that was leftover from some reshuffling